src/shared/components/button/Button.tsx

```diff
@@ -32,203 +20,59 @@ type VariantProps = {
-                        outlineOffset: '1px',
-                    },
-                },
-                destructiveSecondary: {
-                    backgroundColor: `${theme.colors.background}`,
-                    color: `${theme.colors.error}`,
-                    border: `1px solid ${theme.colors.error}`,
-                    '&:hover': {
-                        backgroundColor: `${theme.colors.destructiveSecondary}`,
-                    },
-                    '&:focus': {
-                        boxShadow: `0px 0px 0px 4px ${theme.colors.destructiveSecondaryShadow}`,
-                    },
-                    '&:disabled': {
-                        border: `1px solid ${theme.colors.error200}`,
-                        color: `${theme.colors.error300}`,
-                    },
-                    '&:focus-visible': {
-                        outline: `2px solid ${theme.colors.error700}`,
-                        outlineOffset: '1px',
-                    },
-                },
-            },
-        })};
-`;
-
 export const Button = ({
     iconLeading,
     iconTrailing,
     children,
     isLoading,
+    className,
+    variant,
     ...rest
 }: ButtonProps) => {
     if (isLoading) {
         return (
-            <StyledButton $flexVariant='rowCenter' variant='primary' {...rest}>
+            <button
+                className={cn(
+                    'button-base button-primary',
+                    {
+                        'transition-firefoxSmoothEase focus-visible:outline focus-visible:outline-2':
+                            isFirefox,
+                        'transition-smoothEase': !isFirefox,
+                    },
+                    className,
+                )}
+                {...rest}
+            >
                 <Loader variant='small' />
-            </StyledButton>
+            </button>
         );
     }
 
     return (
-        <StyledButton $flexVariant='rowCenter' {...rest}>
+        <button
+            className={cn(
+                'button-base',
+                {
+                    'transition-firefoxSmoothEase focus-visible:outline focus-visible:outline-2':
+                        isFirefox,
+                    'transition-smoothEase': !isFirefox,
+                    'button-primary': variant === 'primary',
+                    'button-secondary': variant === 'secondary',
+                    'button-secondaryBlack': variant === 'secondaryBlack',
+                    'button-primaryText': variant === 'primaryText',
+                    'button-primaryLink': variant === 'primaryLink',
+                    'button-primaryLinkDestructive': variant === 'primaryLinkDestructive',
+                    'button-linkDestructive': variant === 'linkDestructive',
+                    'button-destructivePrimary': variant === 'destructivePrimary',
+                    'button-destructiveSecondary': variant === 'destructiveSecondary',
```

Comment:
I'd consider moving this into an object outside of this area so you can do `buttonClass[variant]` here instead to clean it up a bit. Then `buttonClass` has all the classnames defined to avoid JIT removing them, but it will make this part more readable